### PR TITLE
[DOCS] Adds machine learning PRs to release notes

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds content from https://github.com/elastic/ml-cpp/blob/7.4/docs/CHANGELOG.asciidoc to the Elasticsearch Release Notes.  It also edits the descriptions of the machine learning PRs that already existed in the release notes.

### CODE REVIEW

**Comment by lcawl** [comment]
@przemekwitek should this PR also be added to the release notes?: https://github.com/elastic/ml-cpp/pull/601

**Comment by droberts195** [comment]
> should this PR also be added to the release notes?: elastic/ml-cpp#601

No, I don't think so, as it was an iteration on #45188 that was made before release - the whole endpoint is new in 7.4.  #45188 is the main feature and should already be in the release notes.  The [docs for that endpoint](https://www.elastic.co/guide/en/elasticsearch/reference/7.4/estimate-memory-usage-dfanalytics.html) reflect the changed field names.

File: docs/reference/release-notes/7.4.asciidoc [hunk 1]
```diff
@@ -9,15 +9,11 @@ Also see <<breaking-changes-7.4,Breaking changes in 7.4>>.
...
-Data Frame::
-* [ML-DataFrame] Combine task_state and indexer_state in _stats {pull}45276[#45276] (issue: {issue}45201[#45201])
-* [ML] Improve response format of data frame stats endpoint {pull}44350[#44350] (issue: {issue}43767[#43767])
-
 Infra/REST API::
 * Update the schema for the REST API specification {pull}42346[#42346] (issue: {issue}35262[#35262])
 
 Machine Learning::
-* [ML] Improve progress reportings for DF analytics {pull}45856[#45856]
+* Improve progress reportings for data frame analytics {pull}45856[#45856]
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Improve progress reporting for data frame analytics {pull}45856[#45856]
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 2]
```diff
@@ -224,19 +214,19 @@ Infra/Settings::
...
-* [ML] Add description to DF analytics {pull}45774[#45774]
-* [ML] Add regression analysis to DF analytics {pull}45292[#45292]
+* Support boolean fields for DF analytics {pull}46037[#46037]
+* Add description to DF analytics {pull}45774[#45774]
+* Add regression analysis to DF analytics {pull}45292[#45292]
 * Introduce formal node ML role {pull}45174[#45174] (issues: {issue}29943[#29943], {issue}43175[#43175])
-* [ML] Improve CSV header row detection in find_file_structure {pull}45099[#45099] (issue: {issue}45047[#45047])
-* [ML] Outlier detection should only fetch docs that have the analyzed … {pull}44944[#44944]
+* Improve CSV header row detection in find_file_structure {pull}45099[#45099] (issue: {issue}45047[#45047])
+* Outlier detection should only fetch docs that have the analyzed … {pull}44944[#44944]
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Outlier detection should only fetch docs that have the analyzed fields {pull}44944[#44944]
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 3]
```diff
@@ -224,19 +214,19 @@ Infra/Settings::
 * Normalize environment paths {pull}45179[#45179] (issue: {issue}45176[#45176])
 
 Machine Learning::
-* [ML] Support boolean fields for DF analytics {pull}46037[#46037]
-* [ML] Add description to DF analytics {pull}45774[#45774]
-* [ML] Add regression analysis to DF analytics {pull}45292[#45292]
+* Support boolean fields for DF analytics {pull}46037[#46037]
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Support boolean fields for data frame analytics {pull}46037[#46037]
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 4]
```diff
@@ -224,19 +214,19 @@ Infra/Settings::
...
+* Add description to DF analytics {pull}45774[#45774]
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Add description to data frame analytics {pull}45774[#45774]
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 5]
```diff
@@ -224,19 +214,19 @@ Infra/Settings::
...
+* Add regression analysis to DF analytics {pull}45292[#45292]
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Add regression analysis to data frame analytics {pull}45292[#45292]
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 6]
```diff
@@ -431,18 +413,22 @@ Infra/Settings::
...
 Machine Learning::
-* [ML] fix two datafeed flush lockup bugs {pull}46982[#46982]
-* [ML] Throw an error when a datafeed needs CCS but it is not enabled for the node {pull}46044[#46044] (issue: {issue}46025[#46025])
+* Fix two datafeed flush lockup bugs {pull}46982[#46982]
+* Throw an error when a datafeed needs CCS but it is not enabled for the node {pull}46044[#46044] (issue: {issue}46025[#46025])
 * Handle "null" value of Estimate memory usage API response gracefully. {pull}45726[#45726] (issue: {issue}44699[#44699])
-* [ML] Remove timeout on waiting for DF analytics result processor to complete {pull}45724[#45724] (issue: {issue}45723[#45723])
-* [ML] Check dest index is empty when starting DF analytics {pull}45094[#45094]
-* [ML] Catch any error thrown while closing data frame analytics process {pull}44958[#44958]
+* Remove timeout on waiting for DF analytics result processor to complete {pull}45724[#45724] (issue: {issue}45723[#45723])
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Remove timeout on waiting for data frame analytics result processor to complete {pull}45724[#45724] (issue: {issue}45723[#45723])
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 7]
```diff
@@ -431,18 +413,22 @@ Infra/Settings::
...
+* Check dest index is empty when starting DF analytics {pull}45094[#45094]
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Check dest index is empty when starting data frame analytics {pull}45094[#45094]
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 8]
```diff
@@ -431,18 +413,22 @@ Infra/Settings::
...
 * Update .ml-config mappings before indexing job, datafeed or df analytics config {pull}44216[#44216] (issue: {issue}44263[#44263])
-* [ML] Wait for .ml-config primary before assigning persistent tasks {pull}44170[#44170] (issue: {issue}44156[#44156])
-* [ML] Fix ML memory tracker lockup when inner step fails {pull}44158[#44158] (issue: {issue}44156[#44156])
-* [ML] Fix datafeed checks when a concrete remote index is present {pull}43923[#43923] (issue: {issue}42113[#42113])
+* Wait for .ml-config primary before assigning persistent tasks {pull}44170[#44170] (issue: {issue}44156[#44156])
+* Fix ML memory tracker lockup when inner step fails {pull}44158[#44158] (issue: {issue}44156[#44156])
+* Fix datafeed checks when a concrete remote index is present {pull}43923[#43923] (issue: {issue}42113[#42113])
+* Rename outlier detection method values `knn` and `tnn` to `distance_kth_nn` and `distance_knn`
+respectively to match the API. {ml-pull}598[#598]
+* Fix occasional (non-deterministic) reinitialisation of modelling for the `lat_long`
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Fix occasional (non-deterministic) reinitialisation of modeling for the `lat_long`
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 9]
```diff
@@ -507,7 +493,21 @@ Snapshot/Restore::
...
+* Fix off-by-one error in checkpoint operations_behind {pull}46235[#46235]
+* Moves failure state transition for MT safety {pull}45676[#45676] (issue: {issue}45664[#45664])
+* Fix _start?force=true bug {pull}45660[#45660]
+* Fix failure state transitions and race condition {pull}45627[#45627] (issues: {issue}45562[#45562], {issue}45609[#45609])
+* Fix starting a batch data frame after stopping at runtime  {pull}45340[#45340] (issues: {issue}44219[#44219], {issue}45339[#45339])
+* Fix null aggregation handling in indexer {pull}45061[#45061] (issue: {issue}44906[#44906])
+* Unify validation exceptions between PUT/_preview {pull}44983[#44983] (issue: {issue}44953[#44953])
+* Treat bulk index failures as an indexing failure {pull}44351[#44351] (issue: {issue}44101[#44101])
+* Prevent task from attempting to run when failed {pull}44239[#44239] (issue: {issue}44121[#44121])
+* Repond with 409 status code when failing _stop {pull}44231[#44231] (issue: {issue}44103[#44103])
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Respond with 409 status code when failing _stop {pull}44231[#44231] (issue: {issue}44103[#44103])
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 10]
```diff
@@ -507,7 +493,21 @@ Snapshot/Restore::
 Task Management::
 * Catch AllocatedTask registration failures {pull}45300[#45300]
 
-
+Data Frame::
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
Transforms::
```

File: docs/reference/release-notes/7.4.asciidoc [hunk 11]
```diff
@@ -507,7 +493,21 @@ Snapshot/Restore::
...
+* Use field_caps API for mapping deduction {pull}46703[#46703] (issue: {issue}46694[#46694])
+* Fix off-by-one error in checkpoint operations_behind {pull}46235[#46235]
+* Moves failure state transition for MT safety {pull}45676[#45676] (issue: {issue}45664[#45664])
+* Fix _start?force=true bug {pull}45660[#45660]
+* Fix failure state transitions and race condition {pull}45627[#45627] (issues: {issue}45562[#45562], {issue}45609[#45609])
+* Fix starting a batch data frame after stopping at runtime  {pull}45340[#45340] (issues: {issue}44219[#44219], {issue}45339[#45339])
+* Fix null aggregation handling in indexer {pull}45061[#45061] (issue: {issue}44906[#44906])
+* Unify validation exceptions between PUT/_preview {pull}44983[#44983] (issue: {issue}44953[#44953])
```
Comment (droberts195) [inline docs/reference/release-notes/7.4.asciidoc]:
```suggestion
* Unify validation exceptions between PUT and _preview {pull}44983[#44983] (issue: {issue}44953[#44953])
```

**Comment by lcawl** [comment]
Thanks @droberts195 I implemented your suggestions via https://github.com/elastic/elasticsearch/commit/0653c95f205d753e969310945f277b912bdf2297 (7.4) and https://github.com/elastic/elasticsearch/commit/66116e39baf13f225187ff6815244b499a4229d5 (7.x)

**Pull request merged**